src/backend/distributed/connection/remote_commands.c

```diff
@@ -277,10 +277,23 @@ ReportConnectionError(MultiConnection *connection, int elevel)
 		messageDetail = pchomp(PQerrorMessage(pgConn));
 	}
 
-	ereport(elevel, (errcode(ERRCODE_CONNECTION_FAILURE),
-					 errmsg("connection error: %s:%d", nodeName, nodePort),
-					 messageDetail != NULL ?
-					 errdetail("%s", ApplyLogRedaction(messageDetail)) : 0));
+	if (messageDetail)
+	{
+		/*
+		 * We don't use ApplyLogRedaction(messageDetail) as we expect any error
+		 * detail that requires log reduction should have done it locally.
+		 */
+		ereport(elevel, (errcode(ERRCODE_CONNECTION_FAILURE),
+						 errmsg("connection failed to the remote node %s:%d with the "
```

Comment:
Grammar-wise, "connection to the remote node %s:%d failed with the following error: %s" seems better. 

Given that the node is not remote in some cases, maybe just "connection to node %s:%d".

Your note on ApplyLogRedaction makes sense.